### PR TITLE
Improve JS tests

### PR DESCRIPTION
**Pull request by appden**
Fixed a number of minor outstanding issues.

### CODE REVIEW

File: tests/js/query-tests.js [hunk 1]
```diff
@@ -77,18 +75,24 @@ function runQuerySuite(suite) {
 
     for (var index in suite.tests) {
         var test = suite.tests[index];
+        var type;
```
Comment (alazier) [inline tests/js/query-tests.js]: why this change? these variables are not used outside of this block. no reason to declare them up here.

**Comment by alazier** [comment]
Everything else looks fine.

File: tests/js/query-tests.js (hunk 1 above)
Comment (appden) [inline tests/js/query-tests.js]: Because `var` is function-scoped, not block scoped, and we can't use `let` in this file since it must be ES5.

**Comment by alazier** [comment]
Why did react-tests Debug fail?

**Comment by appden** [comment]
@alazier no idea, I'm re-running it :smile:

**Comment by appden** [comment]
@alazier I rebased this branch and fixed a bug that was causing debug tests to fail. It's good to go now.

**Comment by alazier** [comment]
👍
